server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -173,11 +173,13 @@ public SnapshotsService(Settings settings, ClusterService clusterService, IndexN
     /**
      * Same as {@link #createSnapshot(CreateSnapshotRequest, ActionListener)} but invokes its callback on completion of
      * the snapshot.
+     * Note: This method is only used in clusters that contain a node older than {@link #NO_REPO_INITIALIZE_VERSION} to ensure a backwards
+     * compatible path for initializing the snapshot in the repository is executed.
      *
      * @param request snapshot request
      * @param listener snapshot completion listener
      */
-    public void executeSnapshot(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
+    public void executeSnapshotLegacy(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
         createSnapshot(request,
```

Comment:
Should this be `createSnapshotLegacy`?
Is this covered by testing?

Reply:
This is covered by mixed cluster tests yes. `createSnapshotLegacy` is already taken for the non-blocking path, I just figured I'd stick with the naming from the original methods for both here.

Reply:
I'm confused. Why is the `executeSnapshotLegacy` method not delegating to `createSnapshotLegacy`?

Reply:
:facepalm: ok now I get it. Sorry. Sadly we don't have a test that would've caught the issue from this mistake and I don't see an easy way to add one. What we'd need is a BwC disruption test where we fail over from new version to old version master while the snapshot is in progress. I'm starting to think we should have that but I don't see a quick way of setting this up since all the BwC tests are rest tests right now and we don't have any way of using the mock repository with those. 

=> Just fixed it now.